lib/src/rollout_evaluator.dart

```diff
@@ -3,347 +3,1008 @@ import 'dart:convert';
+      case UserComparator.containsAnyOf:
+      case UserComparator.notContainsAnyOf:
+        bool negateContainsAnyOf =
+            UserComparator.notContainsAnyOf == comparator;
+        String userAttributeForContains = _getUserAttributeAsString(
+            evaluationContext.key,
+            userCondition,
+            comparisonAttribute,
+            userAttributeValue);
+        return _evaluateContainsAnyOf(
+            negateContainsAnyOf, userCondition, userAttributeForContains);
+      case UserComparator.semverIsOneOf:
+      case UserComparator.semverIsNotOneOf:
+        bool negateSemverIsOneOf =
+            UserComparator.semverIsNotOneOf == comparator;
+        Version userAttributeValueForSemverIsOneOf = _getUserAttributeAsVersion(
+            evaluationContext.key,
+            userCondition,
+            comparisonAttribute,
+            userAttributeValue);
+        return _evaluateSemverIsOneOf(
+            userCondition,
+            userAttributeValueForSemverIsOneOf,
+            negateSemverIsOneOf,
+            comparisonAttribute);
+      case UserComparator.semverLess:
+      case UserComparator.semverLessEquals:
+      case UserComparator.semverGreater:
+      case UserComparator.semverGreaterEquals:
+        Version userAttributeValueForSemverOperators =
+            _getUserAttributeAsVersion(evaluationContext.key, userCondition,
+                comparisonAttribute, userAttributeValue);
+        return _evaluateSemver(userAttributeValueForSemverOperators,
+            userCondition, comparator, comparisonAttribute);
+      case UserComparator.numberEquals:
+      case UserComparator.numberNotEquals:
+      case UserComparator.numberLess:
+      case UserComparator.numberLessEquals:
+      case UserComparator.numberGreater:
+      case UserComparator.numberGreaterEquals:
+        double userAttributeAsDouble = _getUserAttributeAsDouble(
+            evaluationContext.key,
+            userCondition,
+            comparisonAttribute,
+            userAttributeValue);
+        return _evaluateNumbers(userAttributeAsDouble, userCondition,
+            comparator, comparisonAttribute);
+      case UserComparator.isOneOf:
+      case UserComparator.isNotOneOf:
+      case UserComparator.sensitiveIsOneOf:
+      case UserComparator.sensitiveIsNotOneOf:
+        bool negateIsOneOf = UserComparator.sensitiveIsNotOneOf == comparator ||
+            UserComparator.isNotOneOf == comparator;
+        bool sensitiveIsOneOf = UserComparator.sensitiveIsOneOf == comparator ||
+            UserComparator.sensitiveIsNotOneOf == comparator;
+        String userAttributeForIsOneOf = _getUserAttributeAsString(
+            evaluationContext.key,
+            userCondition,
+            comparisonAttribute,
+            userAttributeValue);
+        return _evaluateIsOneOf(
+            userCondition,
+            sensitiveIsOneOf,
+            userAttributeForIsOneOf,
+            _ensureConfigSalt(configSalt),
```

Comment:
Let's require the config salt in the case of sensitive comparators only. (Tip: one way to solve this is to add an `isSensitive` field to the `UserComparator` enum, then use that to determine here if the comparator is sensitive.)

Reply:
True. That was a rookie mistake on my side. Other evaluate methods are affected as well. I will update them.